### PR TITLE
Bump to OTel 1.44.1 and Jenkins 2.479

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Bump to OTel 1.44.1 and Jenkins 2.479

### Testing done

Run project tests

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,12 +24,12 @@
...
         <gitHubRepo>jenkinsci/opentelemetry-api-plugin</gitHubRepo>
-        <opentelemetry.version>1.43.0</opentelemetry.version>
-        <opentelemetry-instrumentation.version>2.9.0</opentelemetry-instrumentation.version>
+        <opentelemetry.version>1.44.1</opentelemetry.version>
+        <opentelemetry-instrumentation.version>2.10.0</opentelemetry-instrumentation.version>
         <opentelemetry-semconv.version>1.28.0-alpha</opentelemetry-semconv.version>
 
         <auto-value.version>1.11.0</auto-value.version>
-        <jenkins.version>2.440.3</jenkins.version>
+        <jenkins.version>2.479.1</jenkins.version>
```
Comment (kuisathaverat) [inline pom.xml:32]:
```suggestion
    <!-- https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/ -->
    <jenkins.baseline>2.479</jenkins.baseline>
    <jenkins.version>${jenkins.baseline}.1</jenkins.version>
```

Comment (kuisathaverat) [reply pom.xml]: you will need to bump the BOM plugin too

```
        <groupId>io.jenkins.tools.bom</groupId>
        <artifactId>bom-${jenkins.baseline}.x</artifactId>
        <version>3654.v237e4a_f2d8da_</version>
```

**Comment by kuisathaverat** [comment]
There are other changes like `StaplerRequest` is now `StaplerRequest2` and `StaplerResponse` is `StaplerResponse2`.
Also, we have to change from `javax.servlet.*` to `jakarta.servlet.`

**Comment by kuisathaverat** [comment]
Also, compile with Java 21

**Review by kuisathaverat — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
-  // Test the minimum required Jenkins Version.
-  [ platform: 'linux', jdk: '11'],
-  [ platform: 'windows', jdk: '17'],
-  [ platform: 'linux', jdk: '21']
+/*
+ See the documentation for more options:
+ https://github.com/jenkins-infra/pipeline-library/
+*/
+buildPlugin(useContainerAgent: false, configurations: [
+  [platform: 'linux', jdk: 21],
```
Comment (jonesbusy) [inline Jenkinsfile:6]: 2.479 only drop support for Java 11. Not 17. 

Might be useful to keep Java 17 build ? (See for the archetype https://github.com/jenkinsci/archetypes/blob/master/common-files/Jenkinsfile)

Comment (cyrille-leclerc) [reply Jenkinsfile:6]: Thanks! https://github.com/jenkinsci/opentelemetry-api-plugin/pull/20

**Pull request merged**